clients/src/main/java/org/apache/kafka/clients/CommonClientConfigs.java

```diff
@@ -63,6 +63,9 @@
     public static final String CLIENT_ID_CONFIG = "client.id";
     public static final String CLIENT_ID_DOC = "An id string to pass to the server when making requests. The purpose of this is to be able to track the source of requests beyond just ip/port by allowing a logical application name to be included in server-side request logging.";
 
+    public static final String CUSTOMIZED_CLIENT_NAME_CONFIG = "customized.client.name";
```

Comment:
dont we traditionally prefix configs that we hotpatch in ?

Reply:
I think "client name" can be confusing. How about using "client software name" as already used in ApiVersionsRequest.java?

Also, let's drop "customized"

Reply:
And should we add another config for version # (see my comment below)?